openml/tasks/task.py

```diff
@@ -42,11 +45,88 @@ def remove_tag(self, tag):
         data = {'task_id': self.task_id, 'tag': tag}
         openml._api_calls._perform_api_call("/task/untag", data=data)
 
+    def _to_dict(self):
+
+        task_container = OrderedDict()
+        task_dict = OrderedDict([
+            ('@xmlns:oml', 'http://openml.org/openml')
+        ])
+        task_container['oml:task_inputs'] = task_dict
+
+        if self.task_id is not None:
+            task_dict['oml:task_id'] = self.task_id
+
+        task_dict['oml:task_type_id'] = self.task_type_id
+
+        source_data = OrderedDict([
+            ('@name', 'source_data'),
+            ('#text', str(self.dataset_id))
+
+        ])
+
+        # not a clustering task
```

Comment:
Hm, could this task-specific code be moved into the actual per-task classes?

Reply:
I understand what you mean. It is doable.